src/OpenLoco/src/World/Station.cpp

```diff
@@ -1055,6 +1055,44 @@ namespace OpenLoco
         }
     }
 
+#pragma warning(push)
+#pragma warning(disable : 4858)
+    // 0x0048F482
+    void removeTileFromStation(const StationId stationId, const World::Pos3& pos, uint8_t rotation)
+    {
+        auto* station = StationManager::get(stationId);
+        auto findPos = pos;
+        findPos.z |= rotation;
+
+        // Bug mitigation: ensure stationTileSize does not exceed the actual array length
+        station->stationTileSize = std::clamp<uint16_t>(station->stationTileSize, 0, static_cast<uint16_t>(std::size(station->stationTiles)));
```

Comment:
I don't think we should do this. Should be handled differently see next comment.